src/pages/index.js

```diff
@@ -21,79 +21,187 @@ import {
 }
 
-const cardListEl = new Section(
-  {
-    items: initialCards,
-    renderer: (data) => {
-      cardListEl.addItem(createCard(data));
-    },
-  },
-
-  ".cards__list"
-);
+function renderItems(items) {
+  items.forEach((item) => {
+    this._renderer(item);
+  });
+}
+api
+  .getInitialCards()
+  .then((cards) => {
+    const cardSection = new Section(
+      {
+        items: cards,
+        renderer: (item) => {
+          const card = createCard(item);
+          cardSection.addItem(card);
+        },
+      },
+      ".cards__list"
+    );
 
-cardListEl.renderItems();
+    cardSection.renderItems();
+  })
+  .catch(console.error);
 
 const userInfo = new UserInfo({
-  nameSelector: ".profile__title",
-  descriptionSelector: ".profile__description",
+  title: ".profile__title",
+  description: ".profile__description",
+  avatarSelector: ".profile__image",
 });
 
+api
+  .getUserInfo()
+  .then((userData) => {
+    userInfo.setUserInfo({
+      title: userData.name,
+      description: userData.about,
+      avatar: userData.avatar,
+    });
+  })
+  .catch((err) => console.error(err));
+
 const cardImageModal = new PopupWithImage("#preview-image-modal");
 cardImageModal.setEventListeners();
 
 function handleImageClick(cardData) {
   cardImageModal.open(cardData);
 }
 
-const addCardPopup = new PopupWithForm(
-  "#add-card-modal",
-  handleAddCardFormSubmit
+const deleteModalConfirmation = new PopupConfirmation(
+  "#modal-confirm",
+  async (cardId, cardElement) => {
+    try {
+      await api.deleteCard(cardId);
+      cardElement.remove();
+    } catch (err) {
+      console.error(`Error On Card Deletion ${err}`);
+    }
+  }
 );
-addCardPopup.setEventListeners();
+
+function handleDeleteCard(card) {
+  deleteModalConfirmation.setSubmitAction(() => {
+    api
+      .deleteCard(card.getId())
+      .then(() => {
+        card.remove();
+        deleteModalConfirmation.close();
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  });
+  deleteModalConfirmation.open();
+}
+
+function handleLikeClick(card) {
+  if (card.setIsLiked) {
+    api
+      .dislikeCard(card._id)
+      .then(() => {
+        card.handleLikeIcon(); // here
+      })
+      .catch(console.error);
+  } else {
+    api
+      .likeCard(card._id)
+      .then(() => {})
+      .catch((err) => {
+        console.error(err);
+      });
+  }
+}
 
 const editProfilePopup = new PopupWithForm(
   "#profile-edit-modal",
-  handleProfileEditSubmit
+  async (data) => {
+    await api.updateUserProfile({
+      name: data.title,
+      about: data.description,
+    });
+    userInfo.setUserInfo({
+      title: data.title,
+      description: data.description,
+    });
+  }
 );
 editProfilePopup.setEventListeners();
 
-function handleProfileEditSubmit(formData) {
-  userInfo.setUserInfo({
-    name: formData.title,
-    description: formData.description,
-  });
-  editProfilePopup.close();
-}
+const cardSection = new Section(
```

Comment:
Only **one instance** of `Section` should be created in the project because there is only **one** section for the cards.